755481 - ocfs2 is not output with hrStorage
757685 - net-snmp ignores reiserfs formatted partitions
748411 - vxfs is not output with hrStorage
846532 - After upgrading net-snmp packages, cvfs filesystem is no longer reported via net-snmp hrstorage
989498 - vzfs missing in hrStorage in an Virtuozzo Container

diff -up net-snmp-5.5/agent/mibgroup/hardware/fsys/fsys_mntent.c.hrStorage-fs net-snmp-5.5/agent/mibgroup/hardware/fsys/fsys_mntent.c
--- net-snmp-5.5/agent/mibgroup/hardware/fsys/fsys_mntent.c.hrStorage-fs	2012-09-14 10:33:46.885860452 +0200
+++ net-snmp-5.5/agent/mibgroup/hardware/fsys/fsys_mntent.c	2012-09-14 10:34:28.812718297 +0200
@@ -136,6 +136,13 @@ _fsys_type( char *typename )
               !strcmp(typename, MNTTYPE_XFS) ||
               !strcmp(typename, MNTTYPE_JFS) ||
               !strcmp(typename, MNTTYPE_NONE) ||
+              !strcmp(typename, MNTTYPE_REISERFS) ||
+              !strcmp(typename, MNTTYPE_OCFS2) ||
+              !strcmp(typename, MNTTYPE_CVFS) ||
+              !strcmp(typename, MNTTYPE_VXFS) ||
+              !strcmp(typename, MNTTYPE_SIMFS) ||
+              !strcmp(typename, MNTTYPE_VZFS) ||
+              !strcmp(typename, MNTTYPE_BTRFS) ||
               !strcmp(typename, MNTTYPE_LOFS))
        return NETSNMP_FS_TYPE_OTHER;
 
diff -up net-snmp-5.5/agent/mibgroup/hardware/fsys/mnttypes.h.hrStorage-fs net-snmp-5.5/agent/mibgroup/hardware/fsys/mnttypes.h
--- net-snmp-5.5/agent/mibgroup/hardware/fsys/mnttypes.h.hrStorage-fs	2012-09-14 10:33:46.832860631 +0200
+++ net-snmp-5.5/agent/mibgroup/hardware/fsys/mnttypes.h	2012-09-14 10:34:46.571658084 +0200
@@ -123,6 +123,27 @@
 #ifndef MNTTYPE_NONE
 #define MNTTYPE_NONE     "none"
 #endif
+#ifndef MNTTYPE_REISERFS
+#define MNTTYPE_REISERFS "reiserfs"
+#endif
+#ifndef MNTTYPE_OCFS2
+#define MNTTYPE_OCFS2    "ocfs2"
+#endif
+#ifndef MNTTYPE_VXFS
+#define MNTTYPE_VXFS    "vxfs"
+#endif
+#ifndef MNTTYPE_CVFS
+#define MNTTYPE_CVFS    "cvfs"
+#endif
+#ifndef MNTTYPE_SIMFS
+#define MNTTYPE_SIMFS   "simfs"
+#endif
+#ifndef MNTTYPE_VZFS
+#define MNTTYPE_VZFS   "vzfs"
+#endif
+#ifndef MNTTYPE_BTRFS
+#define MNTTYPE_BTRFS   "btrfs"
+#endif
 
 /*
  *  File systems to monitor, but not covered by HR-TYPES enumerations
diff -up net-snmp-5.5/agent/mibgroup/hardware/fsys/mounts.h.hrStorage-fs net-snmp-5.5/agent/mibgroup/hardware/fsys/mounts.h
--- net-snmp-5.5/agent/mibgroup/hardware/fsys/mounts.h.hrStorage-fs	2012-09-14 10:33:46.832860631 +0200
+++ net-snmp-5.5/agent/mibgroup/hardware/fsys/mounts.h	2012-09-14 10:37:16.188150792 +0200
@@ -65,6 +65,27 @@
 #ifndef MOUNT_NONE
 #define MOUNT_NONE   "none"
 #endif
+#ifndef MOUNT_REISERFS
+#define MOUNT_REISERFS "reiserfs"
+#endif
+#ifndef MOUNT_OCFS2
+#define MOUNT_OCFS2 "ocfs2"
+#endif
+#ifndef MOUNT_VXFS
+#define MOUNT_VXFS "vxfs"
+#endif
+#ifndef MOUNT_CVFS
+#define MOUNT_CVFS "cvfs"
+#endif
+#ifndef MOUNT_SIMFS
+#define MOUNT_SIMFS "simfs"
+#endif
+#ifndef MOUNT_VZFS
+#define MOUNT_VZFS "vzfs"
+#endif
+#ifndef MOUNT_BTRFS
+#define MOUNT_BTRFS "btrfs"
+#endif
 
 /*
 #ifndef MNTTYPE_AFS
